crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -489,3 +507,21 @@ macro_rules! impl_system_function {
 }
 
 all_tuples!(impl_system_function, 0, 16, F);
+
+pub trait AsSystemLabel<Marker> {
+    fn as_system_label(&self) -> Box<dyn SystemLabel>;
```

Comment:
```suggestion
    fn as_system_label(this: &Self) -> Box<dyn SystemLabel>;
```

I don't think we need this to be a method, for similar reasons we don't need the analagous `IntoSystem::into_system` as a method.

Reply:
Idk if "non method Self functions" is a pattern I want to encourage generally. We had a good reason to do that for IntoSystem: deprecating .system() let us "encourage" users to use the more ergonomic pattern without outright breaking their code. We don't have that context here. Given that methods are the more common pattern, I think the more relevant question is: why _not_ make this a method?

Reply:
Mostly it's never meant to be used by end users, but will clutter up the flyimport suggests when users are attempting to find `.before` or `.after`, especially as alphabetically it is in between those two options.

Reply:
> Mostly it's never meant to be used by end users, 

I'm not sure I agree, given that getting the "closure" label is something you'd want to do with an `as_system_label` call (which would benefit from method-style syntax). Other use cases may emerge in the future too (ex: if something needs to accept a list of labels, we can't pass a bunch of different AsSystemLabel impls ... we need to convert to boxed labels before the call).

I personally don't think autocomplete is much of an issue here, as I think most people type at least the first letter of what they're looking for in autocomplete. `.b` for `.before`, `.a` for `.after` ... `.after` comes before `.as` in an alphabetically sorted list.

Reply:
> I personally don't think autocomplete is much of an issue here, as I think most people type at least the first letter of what they're looking for in autocomplete. .b for .before, .a for .after ... .after comes before .as in an alphabetically sorted list.

Fair enough.